pkg/secrets/pushtofile/secret_group.go

```diff
@@ -75,24 +77,33 @@ func (sg *SecretGroup) pushToFileWithDeps(
 		sg.SecretSpecs,
 	)
 	if err != nil {
-		return err
+		return
 	}
 
 	//// Open and push to file
 	wc, err := depOpenWriteCloser(sg.FilePath, sg.FilePermissions)
 	if err != nil {
-		return err
+		return
 	}
+
 	defer func() {
 		_ = wc.Close()
+		if r := recover(); r != nil {
+			err = fmt.Errorf("panic recovered while executing template with secret values")
+		}
 	}()
 
-	return depPushToWriter(
+	err = depPushToWriter(
 		wc,
 		sg.Name,
 		fileTemplate,
 		secrets,
 	)
+	if err != nil {
+		err = fmt.Errorf("failed to render template for secret group %q with provided secret values", sg.Name)
+		return
```

Comment:
This return can be removed, the code would still work as intended and code climate would be happy.